contributors/devel/development.md

```diff
@@ -341,7 +345,7 @@ To test Kubernetes, you will need to install a recent version of [etcd](https://
 ```sh
 ./hack/install-etcd.sh
 ```
-
+This script requires a minimum bash version of 4.2. 
```

Comment:
@bmnidhin, regarding the minimum bash version requirement – did you find a bash version pinned somewhere in Kubernetes repo?

Rather than hard-coding, could you please add a link to the source of this version information here?
Hard-coded versions in the document will go out-of-date very quickly & adds overhead of maintenance.

Reply:
Fair enough.

The above error string is hardcoded here. https://github.com/kubernetes/kubernetes/blob/e855753ca6e5ec8e061ca05231fcb16b9d5c686c/hack/lib/util.sh#L694

Also there is a similar line in the same doc
https://github.com/kubernetes/community/blob/a3d2ea1e992ba04266bfa0ab25b1316a950fe876/contributors/devel/development.md?plain=1#L354

Reply:
I am still think how better we can handle this. Any suggestions @Priyankasaggu11929 ?

Reply:
My suggestion would be to update the existing `Bash Version Requirement` section to point to [this line in the script](https://github.com/kubernetes/kubernetes/blob/1e827f4b2a46981e4f3056b54b43363e787bbaaa/hack/lib/util.sh#L694) so the text read as following (and then move the section on top of `etcd` section):

```
To successfully run unit tests in Kubernetes, you will need to have at least the minimum required version of bash installed, as specified [here](https://github.com/kubernetes/kubernetes/blob/1e827f4b2a46981e4f3056b54b43363e787bbaaa/hack/lib/util.sh#L694)
```

Because the convention is to enforce checking/verifying versions for any prerequisites as part of the script itself.

Since the bash version check is already there (via sourced file - `hack/install-etcd.sh` -> `hack/lib/init.sh` -> `hack/lib/util.sh`) - so, I think we are ok to drop documenting the bash version requirement here. 


---


> Also there is a similar line in the same doc
> 
> https://github.com/kubernetes/community/blob/a3d2ea1e992ba04266bfa0ab25b1316a950fe876/contributors/devel/development.md?plain=1#L354

Thanks for pointing. I missed reading this section.
But in this case as well, the [same advise was given](https://github.com/kubernetes/community/pull/6713#pullrequestreview-1137922747) but we might have missed that the bash version check was already implemented.